### PR TITLE
Corrections to uart driver for Chibios platform

### PR DESCRIPTION
**Pull request by infinityis**
UART driver for ChibiOS was not working on ARM (testing using STM32F411), it looks like it just didn't follow along with changes in the serial driver as far as how to correctly assign pin modes with palSetLineMode. With these changes, it now works as expected.

I also took the opportunity to refactor some naming to make it more generic (SD instead of SD1) and updated documentation accordingly. A search of the repo showed that no one was actually using the original SD1 naming yet.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
The changes of `SD1` to `SD` should be avoided for now, and at least should be separate from any bug fixes. Renaming currently makes things less consistent, and isnt necessarily aligned to the long term direction.

**Comment by infinityis** [comment]
> The changes of `SD1_` to `SD_` should be avoided for now, and at least should be separate from any bug fixes. Renaming currently makes things less consistent to other ChibiOS drivers, and isnt necessarily aligned to the long term direction.

Noted, renaming has been reverted. Thank you!

**Comment by fauxpark** [comment]
Aligning this code with the split serial driver is not really the right move IMO, if anything the reverse should be done. Ideally, split_common should be using this driver instead of reimplementing it, if possible.

I also note that the I2C driver sets its pins to open drain too so I am not sure why this would be any different - I have not heard of any issues with I2C on F4xx.
https://github.com/qmk/qmk_firmware/blob/master/platforms/chibios/drivers/i2c_master.c#L145-L146

**Comment by infinityis** [comment]
Agreed

> Aligning this code with the split serial driver is not really the right move IMO, if anything the reverse should be done. Ideally, split_common should be using this driver instead of reimplementing it, if possible.
> 
> I also note that the I2C driver sets its pins to open drain too so I am not sure why this would be any different - I have not heard of any issues with I2C on F4xx. https://github.com/qmk/qmk_firmware/blob/master/platforms/chibios/drivers/i2c_master.c#L145-L146

Agreed that split_common could/should use ideally be using the uart driver since is split_common is just a specific use of uart. The uart driver would need to also be modified to include support for half-duplex comms, but conceptually, it all can work.

However, I would disagree that I2C comms are the right comparison to make as far as pin mode settings. Open drain absolutely works for I2C because I2C is indeed an open drain protocol, and so we need to add external pull up resistors to make it work. Half duplex serial comms is also open drain + external pull up resistor, so open drain can work for that too. But full duplex serial comms is pushpull using two dedicated pins with no external pull up resistor. And using push pull matches what is done in the existing serial_usart driver here:
https://github.com/qmk/qmk_firmware/blob/master/platforms/chibios/drivers/serial_usart.c#L200-L201
So if we don't use pushpull mode for uart, then someday when the split_common code starts using the uart driver, all split keyboards that use full duplex comms would stop working unless/until two pull up resistors are added.

**Review by fauxpark — CHANGES_REQUESTED** [review]
Let's keep the GPIOv1 stuff configurable just in case.

**Comment by fauxpark** [comment]
Good point on I2C being open drain, I should have checked the SPI driver too.

File: platforms/chibios/drivers/uart.h [hunk 1]
```diff
@@ -40,23 +40,7 @@
...
-
-#    ifndef SD1_CTS_PAL_MODE
-#        define SD1_CTS_PAL_MODE PAL_MODE_ALTERNATE_OPENDRAIN
-#    endif
-
-#    ifndef SD1_RTS_PAL_MODE
-#        define SD1_RTS_PAL_MODE PAL_MODE_ALTERNATE_OPENDRAIN
-#    endif
-#else
+#if !defined(USE_GPIOV1)
```
[the remark this answers is not in the document]
Comment (infinityis) [reply platforms/chibios/drivers/uart.h]: Sounds good, I've left it configurable now.

**Review by fauxpark — COMMENTED** [review]
Tested on Blackpill F411, looks good. Also checked Bluepill as that uses GPIOv1 - doesn't work either way. Not sure what's up there.

**Review by tzarc — APPROVED** [review]

**Comment by tzarc** [comment]
Guessing F103 doesn't work before this PR either, can probably iterate with another PR later so that at least F4xx is working?

**Review by drashna — APPROVED** [review]

**Comment by xisang233** [comment]
Let me summarize what happend when I tried to make a split keyboard firmware on STM32F103 blue pill board.
1. After I defined `SERIAL_USART_FULL_DUPLEX` `SERIAL_USART_TX_PIN A9` `SERIAL_USART_RX_PIN A10` `SERIAL_USART_DRIVER SD1` like https://docs.qmk.fm/#/serial_driver told me, the compile failed. The error message was `platforms/chibios/drivers/serial_usart.h:11: error: "SERIAL_USART_TX_PIN" redefined [-Werror]`.
I think `USART SD1` equals to A9 as TX pin and A10 as RX pin, so I tried to delect `SERIAL_USART_TX_PIN A9` `SERIAL_USART_RX_PIN A10` and it works.
2. The board cannot connect to each other until I changed `platforms/chibios/drivers/uart.c` and `platforms/chibios/drivers/uart.h`.
Thank you all.

**Comment by tzarc** [comment]
Split serial driver is different to this uart driver.

**Pull request merged**